src/modal-manager-controller.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  LitElement,
+  html,
+  css,
+  customElement,
+  CSSResult,
+  TemplateResult,
+  query,
+  property,
+} from 'lit-element';
+
+import { ModalManagerInterface, ModalManagerMode } from './modal-manager';
+import './modal-manager';
+import { ModalConfig } from './modal-config';
+
+/**
+ * The ModalManagerController is a bridge between the host app and the ModalManager
+ *
+ * It is responsible for handling environmental interactions such as setting up
+ * window resize listeners to account for resizing, as well as setting CSS
+ * properties on the document to disable scrolling when open.
+ *
+ * This may not work for all environments but can be used
+ * as a base class to extend for specific needs.
+ *
+ * @export
+ * @class ModalManagerController
+ * @extends {LitElement}
+ * @implements {ModalManagerInterface}
+ */
+@customElement('modal-manager-controller')
+export class ModalManagerController extends LitElement
+  implements ModalManagerInterface {
+  /**
+   * Custom content to inject into the Modal Manager for passing through slotted content
+   *
+   * @type {TemplateResult}
+   * @memberof ModalManagerController
+   */
+  @property({ type: Object }) slotContent?: TemplateResult;
+
+  /** Reference to the ModalManager DOM element */
+  @query('modal-manager') private modalManager!: ModalManagerInterface;
+
+  /** @inheritdoc */
+  render(): TemplateResult {
+    return html`
+      <modal-manager @modeChanged=${this.modalModeChanged}>
+        ${this.slotContent}
+      </modal-manager>
+    `;
+  }
+
+  /** @inheritdoc */
+  getMode(): ModalManagerMode {
+    return this.modalManager.getMode();
+  }
+
+  /**
+   * Show a modal with a given configuration.
+   *
+   * @param {{
+   *     config: ModalConfig;
+   *     customModalContent?: TemplateResult;
+   *     userClosedModalCallback?: () => void;
+   *   }} options
+   * @returns {Promise<void>}
+   * @memberof ModalManagerController
+   */
+  async showModal(options: {
+    config: ModalConfig;
+    customModalContent?: TemplateResult;
+    userClosedModalCallback?: () => void;
+  }): Promise<void> {
+    return this.modalManager.showModal(options);
+  }
+
+  /**
+   * Close the modal
+   *
+   * @returns {void}
+   * @memberof ModalManagerController
+   */
+  closeModal(): void {
+    return this.modalManager.closeModal();
+  }
+
+  /** @inheritdoc */
+  firstUpdated(): void {
+    window.addEventListener('resize', () => {
```

Comment:
This should be throttled so there are far fewer calls while drag-resizing.

Additionally, is there a teardown method that removes this listener?

Reply:
This is throttled now.